### ISSUE TITLE
Problem with box_dots and numeric keys

### ISSUE THREAD

**Issue by richieadler** [cdgriffith/Box#121]
Windows 10, Python 3.6, Box 4.0.2:

```
>>> info = box.Box.from_yaml('uno:\n  2: tres', default_box=True, box_dots=True)
Traceback (most recent call last):
  File "<interactive input>", line 1, in <module>
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 685, in from_yaml
    return cls(data, **box_args)
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 224, in __init__
    self.__setitem__(k, v)
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 442, in __setitem__
    self.__convert_and_store(key, value)
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 377, in __convert_and_store
    value = self.__class__(value, __box_heritage=(self, item), **self.__box_config())
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 224, in __init__
    self.__setitem__(k, v)
  File "C:\Users\UserName\AppData\Local\Programs\Python\Python36-32\lib\site-packages\box\box.py", line 436, in __setitem__
    if self._box_config['box_dots'] and '.' in key:
TypeError: argument of type 'int' is not iterable
```

**Comment by cdgriffith** [comment]
Sorry you are running into these, but thanks for reporting all the new feature's bugs!

**Comment by cdgriffith** [comment]
Fixed in 4.0.3
